tests/core/test_config.py

```diff
@@ -593,6 +593,20 @@ async def test_raw_with_partial_primary_keys(config):
     assert await config.custom("CUSTOM", "primary_key").identifier() is False
 
 
+@pytest.mark.asyncio
+async def test_cast_subclass_default(config):
```

Comment:
I think an issue reference for more context would make sense here:
```suggestion
async def test_cast_subclass_default(config):
    # regression test for GH-5557/GH-5585
```